src/test/java/uk/gov/hmcts/reform/sscs/bulkscancore/ccd/CaseDataHelperTest.java

```diff
@@ -0,0 +1,157 @@
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+import uk.gov.hmcts.reform.sscs.common.SampleCaseDataCreator;
+
+@RunWith(SpringRunner.class)
+public class CaseDataHelperTest {
+
+    private static final String EVENT_TOKEN =
+        "eyJhbGciOiJIUzI1NiJ9.eyJqdGkiOiI4cG1jN284YWdyaDA0MzIxcGJ1ZG5rOGs1cCIsInN1YiI6IjI1IiwiaWF0IjoxNTM4OTk2MTQyLCJldm"
+            + "VudC1pZCI6ImFwcGVhbENyZWF0ZWQiLCJjYXNlLXR5cGUtaWQiOiJCZW5lZml0IiwianVyaXNkaWN0aW9uLWlkIjoiU1NDUyIsImNhc2U"
+            + "tdmVyc2lvbiI6ImJmMjFhOWU4ZmJjNWEzODQ2ZmIwNWI0ZmEwODU5ZTA5MTdiMjIwMmYifQ.v5CIHOFPKCrxjhEKAPDhltd-ErynobIY5"
+            + "FVjXJffEVU";
+
+    private static final Long CASE_ID = Long.valueOf("1538992487551266");
+
+    private static final String INVALID_SERVICE_AUTH = "invalidserviceauth";
+
+    @Mock
+    private CoreCaseDataApi coreCaseDataApi;
+
+    private CaseDataHelper caseDataHelper;
+
+    private SampleCaseDataCreator caseDataCreator = new SampleCaseDataCreator();
+
+    @Before
+    public void setUp() {
+        caseDataHelper = new CaseDataHelper(coreCaseDataApi);
+        ReflectionTestUtils.setField(caseDataHelper, "jurisdictionId", "SSCS");
+        ReflectionTestUtils.setField(caseDataHelper, "caseType", "Benefit");
+        ReflectionTestUtils.setField(caseDataHelper, "eventId", "appealCreated");
+    }
+
+    @Test
+    public void should_create_case_in_ccd_and_return_case_id() {
+        // Given
+        when(coreCaseDataApi.startForCaseworker(
+            TEST_USER_AUTH_TOKEN,
+            TEST_SERVICE_AUTH_TOKEN,
+            TEST_USER_ID,
+            "SSCS",
+            "Benefit",
+            "appealCreated")
+        ).thenReturn(StartEventResponse.builder()
+            .token(EVENT_TOKEN)
+            .build());
+
+        when(coreCaseDataApi.submitForCaseworker(
+            TEST_USER_AUTH_TOKEN,
+            TEST_SERVICE_AUTH_TOKEN,
+            TEST_USER_ID,
+            "SSCS",
+            "Benefit",
+            true,
+            sscsCaseDataContent())
+        ).thenReturn(CaseDetails.builder()
+            .id(CASE_ID)
+            .data(caseDataCreator.sscsCaseData())
+            .createdDate(LocalDateTime.of(2018, 1, 1, 0, 0))
+            .build());
+
+        // When
+        Long actualCaseId = caseDataHelper.createCase(
+            caseDataCreator.sscsCaseData(),
+            TEST_USER_AUTH_TOKEN,
+            TEST_SERVICE_AUTH_TOKEN,
+            TEST_USER_ID
+        );
+
+        // Then
+        assertThat(actualCaseId).isEqualTo(CASE_ID);
+    }
+
+    @Test
+    public void should_throw_forbidden_exception_when_service_auth_is_invalid() {
+        // Given
+        when(coreCaseDataApi.startForCaseworker(
+            TEST_USER_AUTH_TOKEN,
+            INVALID_SERVICE_AUTH,
+            TEST_USER_ID,
+            "SSCS",
+            "Benefit",
+            "appealCreated")
+        ).thenThrow(new HttpClientErrorException(HttpStatus.FORBIDDEN));
```

Comment:
how do you know this is the exception that will be thrown out when a 403 happens?

Reply:
To be honest I am not sure and hence I have added in the description that all the scenarios will be covered in functional tests where nothing is mocked. Added unit tests just to cover basic scenarios and coverage for now.

Reply:
I would suggest not adding this test as I suspect its wrong, at the integration test level this should probably be covered by wiremock?

Reply:
ok makes sense will cover it in integration tests